### PR TITLE
Update babel mcfly

### PR DESCRIPTION
**Pull request by jcperez-ch**
Few changes made to the source code:

- Replace vars by `const` and `let`.
- Upgrade all dev and no dev dependencies to latest version.
- Get rid of react-tools dev dependency.
- Update the example in order to work with `npm start` correctly.

Introducing React 15.0 so we can implement `React` and `ReactDOM` separately.

@kenwheeler I hope you enjoy merging this PR ;)

### CODE REVIEW

**Comment by kenwheeler** [comment]
This is lovely, thank you

File: example/js/flux/mcFly.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-var McFly = require('mcfly');
+var McFly = require('../../../index.js');
```
Comment (jcperez-ch) [inline example/js/flux/mcFly.js]: Here you go guys... it works like a charm :ok_hand: 

@kenwheeler @tomatau

**Comment by kenwheeler** [comment]
Hey, before I cut a release, can you update jest to latest, add babel jest and a babelrc and make these tests pass?

**Comment by jcperez-ch** [comment]
Sure @kenwheeler

**Pull request merged**